rust/extractor/src/qltest.rs

```diff
@@ -0,0 +1,84 @@
+use crate::config::Config;
+use anyhow::Context;
+use glob::glob;
+use itertools::Itertools;
+use log::info;
+use std::fs;
+use std::process::Command;
+
+fn dump_lib() -> anyhow::Result<()> {
+    let path_iterator = glob("*.rs").context("globbing test sources")?;
+    let paths = path_iterator
+        .collect::<Result<Vec<_>, _>>()
+        .context("fetching test sources")?;
+    let lib = paths
+        .iter()
+        .filter(|p| !["lib.rs", "main.rs"].contains(&p.file_name().unwrap().to_str().unwrap()))
```

Comment:
Try to avoid `unwrap` when dealing with file paths, especially `to_str`. While `to_str` is unlikely to fail, it can in case of files with badly encoded names or in case of bad locale settings.  In this case there is no need for `to_str`, you could just make `OsString`s of `["lib.rs", "main.rs"]`. You can also drop files with bad names and issue a warning.